### PR TITLE
Storage module impro perf

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
- Improve test env script => add cleanup function
- Improve `Vec<u8>` conversion in `set_chunk`
- Improve `set_manifest` => allow to do operation in cache
- Improve `get_manifest` => allow to do operation in cache
- Improve `load_file_descriptor` => allow to do operation in cache

### CODE REVIEW

File: src/storage/workspace_storage.rs [hunk 1]
```diff
@@ -266,16 +262,24 @@ impl WorkspaceStorage {
...
+    fn set_chunk(
+        &self,
+        py: Python,
+        chunk_id: ChunkID,
+        block: PyObject,
+    ) -> PyResult<FutureIntoCoroutine> {
+        let block = match block.extract::<&PyByteArray>(py) {
+            Ok(x) => x.to_vec(),
+            Err(_) => block.extract::<&PyBytes>(py)?.as_bytes().to_vec(),
+        };
```
Comment (vxgmichel) [inline src/storage/workspace_storage.rs:274]: This should be done using `BytesWrapper` et `unwrap_bytes!` instead

File: tests/scripts/run_testenv.sh [hunk 2]
```diff
@@ -41,6 +41,21 @@ $SCRIPT_DIR/run_testenv.py --source-file $SOURCE_FILE $@ || return $?
 source $SOURCE_FILE
 
 # Clean up
-rm $SOURCE_FILE
-unset SOURCE_FILE
-unset SCRIPT_DIR
```
Comment (vxgmichel) [inline tests/scripts/run_testenv.sh:46]: If those variables stay in the current shell we should find less conflicting names for them

**Review by vxgmichel — APPROVED** [review]
A couple of comments, LGTM otherwise :+1:

File: src/storage/workspace_storage.rs (hunk 1 above)
Comment (FirelightFlagboy) [reply src/storage/workspace_storage.rs:274]: Would be done when I rebase the base branch with master

File: tests/scripts/run_testenv.sh (hunk 2 above)
Comment (FirelightFlagboy) [reply tests/scripts/run_testenv.sh:46]: They may stay in the current shell if the user don't invoke `cleanup` after the test are finished.
I've renamed them by adding the prefix `PARSEC_TESTENV_`

**Review by TimeEngineer — APPROVED** [review]

**Pull request merged**